<launch>
  <!-- launching arguments passed as param for main task -->
  <arg name="side_number" default="3"/>
  <arg name="side_length" default="5"/>
  <arg name="dir_counter_clock" default="true"/>

  <!-- gazebo launch node -->
  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_empty_world.launch"/>

  <!-- main task node -->
  <node pkg="simple_shapes" type="simple_shapes_drawer" name="simple_shapes_drawer" output="screen">
    <param name="side_number" value="$(arg side_number)"/>
    <param name="side_length" value="$(arg side_length)"/>
    <param name="dir_counter_clock" value="$(arg dir_counter_clock)"/>
    <rosparam file="$(find simple_shapes)/param/simple_shapes.yaml" command="load"/>
  </node>

  <!-- rviz node -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find simple_shapes)/config/simple_shape.rviz" />

</launch>
